flink-scala/src/main/scala/org/apache/flink/api/scala/typeutils/OptionSerializer.scala

```diff
@@ -65,7 +65,7 @@ class OptionSerializer[A](val elemSerializer: TypeSerializer[A])
     case Some(a) =>
       target.writeBoolean(true)
       elemSerializer.serialize(a, target)
-    case None =>
+    case _ =>
```

Comment:
The other way round is more likely, i.e., having data serialized with the old version (v1.1) and deserializing with the new one (v1.2). That would work without problems.

It is of course possible, that a user writes with v1.2 and tries to read with v1.1 but that is a) IMO, rather unlikely and b) I'm not sure whether we need to guard for that and c) the user would probably notice, because the deserialization will be broken as you noticed (which is better than a silent corruption). But let's maybe check what others think. 

@StephanEwen, do you think we should support `null` for Scala `Option` which would mean to change the serialization logic as discussed above?